packages/shared/src/hooks/useExitConfirmation.ts

```diff
@@ -0,0 +1,52 @@
+import { useRouter } from 'next/router';
+import { useEffect, useRef } from 'react';
+
+interface ExitProps {
+  message?: string;
+  onValidateAction: () => boolean;
+}
+
+export interface UseExitConfirmation {
+  onAskConfirmation: (value: boolean) => void;
+}
+
+export function useExitConfirmation({
+  message = 'You have unsaved changes that will be lost if you leave the page',
+  onValidateAction,
+}: ExitProps): UseExitConfirmation {
+  const confirmRef = useRef(true);
+  const router = useRouter();
+
+  useEffect(() => {
+    if (!router.isReady) return null;
+
+    const closeHandler = (e: BeforeUnloadEvent) => {
+      if (!confirmRef.current || onValidateAction()) return;
```

Comment:
would move ti to variable `shouldAskConfirmation` and then check `if(shouldAskConfirmation)`, just so we know what this does without looking at inner workings of `beforeunload` and `returnValue`.

Reply:
Makes sense. Let me add that.